### ISSUE TITLE
Named Resource for sub application are not resolvable

### ISSUE THREAD

**Issue by ediskandarov-cur** [aio-libs/aiohttp#1489]
## Long story short

Reverse URL Constructing using Named Resources doesn't work for sub applications. 

## Expected behaviour

Named Resource for sub application should be resolvable:

```
request.app.router['subapp-resource'].url_for()
```

## Actual behaviour

```
Error handling request
Traceback (most recent call last):
  File "/Users/e.iskandarov/.virtualenvs/chess-server/lib/python3.5/site-packages/aiohttp/web_server.py", line 61, in handle_request
    resp = yield from self._handler(request)
  File "/Users/e.iskandarov/.virtualenvs/chess-server/lib/python3.5/site-packages/aiohttp/web.py", line 249, in _handle
    resp = yield from handler(request)
  File "/Users/e.iskandarov/.virtualenvs/chess-server/lib/python3.5/site-packages/aiohttp_jinja2/__init__.py", line 89, in wrapped
    context = yield from coro(*args)
  File "/Users/e.iskandarov/aiogames/aiogames/apps/index/views.py", line 6, in index
    chess_board_url = request.app.router['chess-board'].url_for()
  File "/Users/e.iskandarov/.virtualenvs/chess-server/lib/python3.5/site-packages/aiohttp/web_urldispatcher.py", line 751, in __getitem__
    return self._named_resources[name]
KeyError: 'chess-board'
```

## Steps to reproduce

Create a project with sub application and try to resolve it in view.

## Your environment

macOS, Python 3.5.2, aiohttp 0.1.2

**Comment by asvetlov** [comment]
It's intended.
If you main app should do URL reversing for subapp it always could use the following explicit technique:

```
sub_app = web.Application()
app.router.add_subapp('/prefix', sub_app)
app['subapp'] = subapp


async def handler(request):  # main app's handler
    subapp = request.app['subapp']
    url = subapp.router['chess-board'].url_for()
```

**Comment by ediskandarov-cur** [comment]
Got it. Thanks!

**Comment by asvetlov** [comment]
Looks like this behavior should be explicitly mentioned.
Would you provide a Pull Request?

**Comment by asvetlov** [comment]
I mean adding a paragraph into web.rst

**Comment by ediskandarov-cur** [comment]
Ok, will prepare PR tonight.

**Comment by asvetlov** [comment]
Cool!

**Comment by asvetlov** [comment]
Fixed by #1496
